### ISSUE TITLE
FlowableOrderedMerge for ParallelFlowables

### ISSUE THREAD

**Issue by swimmesberger** [akarnokd/RxJava2Extensions#16]
It would be nice to have a orderedMerge method which can be used with ParallelFlowables if I'm correct the FlowableOrderedMerge could be easily adjusted to provide such functionality by simply changing the constructor from:
```java
    FlowableOrderedMerge(Publisher<T>[] sources, Iterable<? extends Publisher<T>> sourcesIterable,
            Comparator<? super T> comparator,
            boolean delayErrors, int prefetch) {
```
to
```java
    FlowableOrderedMerge(ParallelFlowable<T> source,
            Comparator<? super T> comparator,
            boolean delayErrors, int prefetch) {
```

and changing the subscribeActual method to:
```java
    @Override
    protected void subscribeActual(Subscriber<? super T> s) {
        final MergeCoordinator<T> parent = new MergeCoordinator<T>(s, comparator, this.source.parallelism(), prefetch, delayErrors);
        s.onSubscribe(parent);
        this.source.subscribe(parent.subscribers);
    }
```
I don't know how you want to integrate the change to support both cases but it would be very useful to have :)

**Comment by akarnokd** [comment]
There is the [sorted](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/parallel/ParallelFlowable.java#L453) operator.

Otherwise, PR welcome.

**Comment by swimmesberger** [comment]
Yeah but the sorted operator only works on finite streams and works quite differently than the FlowableOrderedMerge operator. I have tested the changed FlowableOrderedMerge in my application with ParallelFlowables and it works like a charm :)

**Comment by akarnokd** [comment]
You can post a PR that adds it to [ParallelFlowables](https://github.com/akarnokd/RxJava2Extensions/blob/master/src/main/java/hu/akarnokd/rxjava2/parallel/ParallelTransformers.java) then.

**Comment by swimmesberger** [comment]
Merged with #17
